### PR TITLE
Use SameValue in verifyProperty helper

### LINKED ISSUE #1146: Improvements for the verifyProperty helper
2 suggestions:

On https://github.com/tc39/test262/blob/master/harness/propertyHelper.js#L42-L43

It should use the assert.sameValue to properly compare NaN as well.

Also, it still needs implementation to verify `get` and `set` values in accessor descriptors, those won't need to use assert.sameValue.

cc @rwaldron, you might be interested on this.

### PR DESCRIPTION
**Pull request by shvaikalesh**
Closes #1146.

### CODE REVIEW

File: harness/propertyHelper.js [hunk 1]
```diff
@@ -73,6 +73,18 @@ function verifyProperty(obj, name, desc, options) {
     }
   }
 
+  if (Object.prototype.hasOwnProperty.call(desc, 'get')) {
```
Comment (shvaikalesh) [inline harness/propertyHelper.js]: This one is not quite useful.

There are ~30 accessors in spec.
All of them are non-enumerable, configurable, and lack `set` (well, except for `Object#__proto__`).
We can introduce a new helper to check their descriptors:

```js
verifyGetterProperty(Map.prototype, 'size');
```

File: harness/propertyHelper.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ function verifyProperty(obj, name, desc, options) {
   var failures = [];
 
   if (Object.prototype.hasOwnProperty.call(desc, 'value')) {
-    if (desc.value !== originalDesc.value) {
+    if (!assert._isSameValue(desc.value, originalDesc.value)) {
```
Comment (leobalter) [inline harness/propertyHelper.js]: this one depends on `assert` having a public property `_isSameValue` which not documented and might break if any runner uses a custom harness.

I'd be fine to actually expand this code here to verify for `NaN` cases, but still avoiding an abstraction that is not in this file.

File: harness/propertyHelper.js (hunk 1 above)
Comment (leobalter) [reply harness/propertyHelper.js]: I'd prefer deferring this change to when we have usage and tests for it. It's pretty tricky to check/compare the value of the get functions and we might end up not having a proper expected value.

File: harness/propertyHelper.js [hunk 3]
```diff
@@ -73,6 +73,18 @@ function verifyProperty(obj, name, desc, options) {
...
+    if (desc.get !== originalDesc.get) {
+      failures.push('descriptor get should be ' + desc.get);
+    }
+  }
+
+  if (Object.prototype.hasOwnProperty.call(desc, 'set')) {
+    if (desc.set !== originalDesc.set) {
+      failures.push('descriptor set should be ' + desc.set);
+    }
+  }
```
Comment (leobalter) [inline harness/propertyHelper.js]: same for set, better to defer this until we write tests using it.

Comment (shvaikalesh) [reply harness/propertyHelper.js]: 5ac819f

File: harness/propertyHelper.js (hunk 2 above)
Comment (shvaikalesh) [reply harness/propertyHelper.js]: 4f4d2d6

File: harness/propertyHelper.js (hunk 1 above)
Comment (shvaikalesh) [reply harness/propertyHelper.js]: 6935757

**Pull request merged**